src/tools/start_quest_tool.py

```diff
@@ -98,8 +99,14 @@ def start_quest(  # noqa: C901
 
         return quest
 
-    def num_problems_to_encounter(self, difficulty_base: int) -> int:
-        return (difficulty_base // 4) + 2 + randint(0, 2)  # noqa: S311
+    def num_problems_to_encounter(
+        self, difficulty_base: int, server_settings: ServerSettings
+    ) -> int:
+        return (
+            math.floor(difficulty_base * server_settings.problems_per_quest_scale)
+            + server_settings.min_problems_per_quest
+            + randint(0, server_settings.max_additional_problems_per_quest)
```

Comment:
question(non-blocking): does pre-commit not complain here about pseudo-random values?

it feels like this may be missing a `# noqa: S311`, but maybe this is smarter ?

Reply:
I've never seen that one